substrate/primitives/src/networks.rs

```diff
@@ -10,38 +10,98 @@ use borsh::{BorshSerialize, BorshDeserialize};
 use serde::{Serialize, Deserialize};
 
 use sp_core::{ConstU32, bounded::BoundedVec};
+use sp_std::{vec, vec::Vec};
 
 #[cfg(feature = "borsh")]
 use crate::{borsh_serialize_bounded_vec, borsh_deserialize_bounded_vec};
 
-/// The type used to identify networks.
+/// The type used to identify external networks.
 #[derive(
   Clone, Copy, PartialEq, Eq, Hash, Debug, Encode, Decode, PartialOrd, Ord, MaxEncodedLen, TypeInfo,
 )]
 #[cfg_attr(feature = "std", derive(Zeroize))]
 #[cfg_attr(feature = "borsh", derive(BorshSerialize, BorshDeserialize))]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-pub enum NetworkId {
-  Serai,
+pub enum ExternalNetworkId {
   Bitcoin,
   Ethereum,
   Monero,
 }
+
+/// The type used to identify networks.
+#[derive(
+  Clone, Copy, PartialEq, Eq, Hash, Debug, Encode, Decode, PartialOrd, Ord, MaxEncodedLen, TypeInfo,
```

Comment:
We shouldn't derive Encode/Decode/MaxEncodedLen. This is now a one/two-byte encoding as there's one byte used for Serai/External, and one byte used for the ExternalNetworkId. We can manually implement it such that Serai == 0, ExternalNetworkId = id.encode()[0] + 1.

We can also manually implement it for ExternalNetworkId so that ENId starts at 1. This would preserve the existing encoding format and allow upgrading ENId instances to NId instances in the future, without changing the wire format/invalidating storage (though we would need a clause banning using the ::Serai variant prior to some block height if we ever introduced such an expansion). In that case, the NId encoding would simply be `if Serai [0] else enid.encode()`.

Reply:
Second option does seem to make sense to me. Should we do same for the `Coin` and `ExternalCoin` types?